src/bindings/python/src/pyopenvino/graph/model.cpp

```diff
@@ -327,19 +327,47 @@ void regclass_graph_Model(py::module m) {
     model.def(
         "reshape",
         [](ov::Model& self, const py::list& partial_shape) {
-            ov::PartialShape new_shape(Common::partial_shape_from_list(partial_shape));
-            py::gil_scoped_release release;
-            self.reshape(new_shape);
+            if (py::len(partial_shape) > 0 && py::isinstance<py::list>(partial_shape[0])) {
+                if (py::len(partial_shape) != self.inputs().size()) {
+                    throw py::value_error("Reshape failed due to mismatched lengths");
+                }
+                std::map<ov::Output<ov::Node>, ov::PartialShape> shapes_map;
+
+                for (size_t i = 0; i < py::len(partial_shape); ++i) {
+                    py::list shape_list = partial_shape[i].cast<py::list>();
+                    ov::PartialShape shape = Common::partial_shape_from_list(shape_list);
+                    shapes_map[self.input(i)] = shape;
+                }
+
+                py::gil_scoped_release release;
+                self.reshape(shapes_map);
+            } else {
+                ov::PartialShape new_shape(Common::partial_shape_from_list(partial_shape));
+                py::gil_scoped_release release;
+                self.reshape(new_shape);
+            }
         },
         py::arg("partial_shape"),
         R"(
                 Reshape model input.
 
                 GIL is released while running this function.
 
-                :param partial_shape: New shape.
-                :type partial_shape: list
+                :param partial_shape: New shape or list of shapes.
+                :type partial_shape: Union[List[int], List[List[int]]]
```

Comment:
Should we allow the following types to represent `PartialShape`?

            (1) `openvino.runtime.PartialShape`
            (2) `list` consisting of dimensions
            (3) `tuple` consisting of dimensions
            (4) `str`, string representation of `openvino.runtime.PartialShape`

Reply:
didn't get what you mean, do you mean something like this to be also included in this ticket? 

model.reshape([(1,2), (2,3), [4,5]])

Reply:
@amkarn258 @rkazants means to extend the docstring with all possible list's element types.

Possible types of list elements you may found [here](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/src/pyopenvino/core/common.cpp#L547).
[Example ](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/src/pyopenvino/graph/model.cpp#L419) of such a docstring

Reply:
@amkarn258 meanwhile I'd be very great if you extend also your test with these types